.github/workflows/build-and-test.yml

```diff
@@ -5,115 +5,185 @@ on:
+    # lower the number of agents to reduce spent compute time.
     strategy:
       matrix:
-        agent: [1, 2, 3]
+        agent: [1, 2, 3, 4]
 
     steps:
-      - uses: actions/checkout@v2
-      - uses: ./.github/actions/setup
+      - name: Checkout all commits
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up dependencies
+        uses: ./.github/actions/setup
 
-      - name: Start Nx Agent ${{ matrix.agent }}
+      - name: Start Nx Cloud DTE Agent ${{ matrix.agent }}
         run: yarn nx-cloud start-agent
 
-  build:
+  # We're using Nx Cloud for Distributed Task Execution
+  # Reference: https://nx.dev/using-nx/dte
+  #
+  # The coordinator outputs the combination of task outputs from the agents,
+  # both terminal and file outputs
+  dte_coordinator:
+    name: DTE Coordinator
     runs-on: ubuntu-latest
+    env:
+      NX_CLOUD_DISTRIBUTED_EXECUTION: true
+      NX_DISTRIBUTED_TASK_EXECUTION: true
 
     steps:
-      - uses: actions/checkout@v2
-      - uses: ./.github/actions/setup
-
-      - run: yarn nx-cloud start-ci-run
-
+      - name: Checkout all commits
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up dependencies
+        uses: ./.github/actions/setup
+
+      - name: Start Nx Cloud DTE
+        run: yarn nx-cloud start-ci-run --stop-agents-after="e2e" --agent-count=4
+
+      - name: Lint Nx workspace
+        env:
+          NX_CLOUD_DISTRIBUTED_EXECUTION: false
+          NX_DISTRIBUTED_TASK_EXECUTION: false
+        run: yarn nx workspace-lint
+      - name: Check formatting
+        env:
+          NX_CLOUD_DISTRIBUTED_EXECUTION: false
+          NX_DISTRIBUTED_TASK_EXECUTION: false
+        run: yarn nx format:check
+
+      # Distribution strategy for 2 vCPUs per hosted runner (GitHub Free):
+      #
+
+      # lint: 2 tasks assigned at a time, 1 task per vCPU
       - name: Run Affected lint
-        run: yarn nx affected:lint --parallel --maxParallel=3
+        run: yarn nx affected:lint --parallel --max-parallel=2 --quiet
 
+      # test: 1 task assigned at a time with 2 parallel processes, 1 process per vCPU
       - name: Run Affected test
-        run: yarn nx affected:test --parallel --maxParallel=3
+        id: test
+        run: yarn nx affected:test --parallel --max-parallel=1 --max-workers=2 --ci --code-coverage
 
-      - name: Run Affected e2e
-        run: yarn nx affected:e2e
-
-      # don't run build in parallel, can cause dead locks
+      # build: 2 tasks assigned at a time, 1 task per vCPU
       - name: Run Affected build
-        run: yarn nx affected:build --exclude=docs --parallel=false
+        run: yarn nx affected:build --exclude=docs --parallel --max-parallel=2
+
+      # e2e: 1 tasks assigned at a time, 1 process per agent
+      - name: Run Affected e2e
+        run: yarn nx affected:e2e --parallel=false --max-parallel=1
```

Comment:
isn't max-parallel=1 === parallel=false?

Reply:
I'm not sure. I'm used to more recent versions of Nx where `--max-parallel=<n>` is deprecated in favor of `--parallel=<n>`.

Reply:
I guess since `parallel` is set to `false` in `nx.json`, tasks are not run in parallel by default.